File: code/appspython/implement/Converters/test/tConvert.py
import numarray as NUM

from _tConvert import *

t = tConvert();

print t.testbool (True);
print t.testbool (False);
print t.testint (-1);
print t.testint (10);
print t.testfloat (3.14);
print t.testfloat (12);
print t.teststring ("this is a string");

print t.testvecint ([1,2,3,4]);
print t.testveccomplex ([1+2j, -1-3j, -1.5+2.5j]);
print t.testvecstr (["a1","a2","b1","b2"])
print t.testipos ([2,3,4]);

print t.testvh (True);
print t.testvh (2);
print t.testvh (1.3);
print t.testvh (10-11j);
print t.testvh ("str");
print t.testvh ([True]);
print t.testvh ([2,4,6,8,10]);
print t.testvh ([1.3,4,5,6]);
print t.testvh ([10-11j,1+2j]);
print t.testvh (["str1","str2"]);
print t.testvh ({"shape":[2,2],"array":["str1","str2","str3","str4"]});
a  =  t.testvh ({"shape":[2,2],"array":["str1","str2","str3","str4"]});
print a;
print t.testvh (a);

a  =  t.testvh ([10-11j,1+2j]);
print a.shape;
print a.type();
print t.testvh (a);

b  =  NUM.array([[2,3],[4,5]]);
print b;
print t.testvh (b);

File: code/appspython/implement/Converters/test/tConvert.cc
//# pycasatable.cc: python module for AIPS++ table system
//# Copyright (C) 2006
//# Associated Universities, Inc. Washington DC, USA.
//#
//# This library is free software; you can redistribute it and/or modify it
//# under the terms of the GNU Library General Public License as published by
//# the Free Software Foundation; either version 2 of the License, or (at your
//# option) any later version.
//#
//# This library is distributed in the hope that it will be useful, but WITHOUT
//# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
//# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Library General Public
//# License for more details.
//#
//# You should have received a copy of the GNU Library General Public License
//# along with this library; if not, write to the Free Software Foundation,
//# Inc., 675 Massachusetts Ave, Cambridge, MA 02139, USA.
//#
//# Correspondence concerning AIPS++ should be addressed as follows:
//#        Internet email: aips2-request@nrao.edu.
//#        Postal address: AIPS++ Project Office
//#                        National Radio Astronomy Observatory
//#                        520 Edgemont Road
//#                        Charlottesville, VA 22903-2475 USA
//#
//# $Id: tConvert.cc,v 1.3 2006/09/29 02:46:59 gvandiep Exp $

#include <appspython/Converters/PycExcp.h>
#include <appspython/Converters/PycBasicData.h>
#include <appspython/Converters/PycValueHolder.h>
#include <appspython/Converters/PycRecord.h>
#include <appspython/Converters/PycArray.h>
#include <casa/Arrays/ArrayIO.h>

#include <boost/python.hpp>

using namespace boost::python;

namespace casa { namespace appspython {

  struct TConvert
  {
    TConvert() {}
    Bool  testbool (Bool in)
      {cout << "bool " << in << endl; return in;}
    Int testint (Int in)
      {cout << "Int " << in << endl; return in;}
    Double testfloat (Float in)
      {cout << "Float " << in << endl; return in;}
    DComplex testcomplex (const DComplex& in)
      {cout << "DComplex " << in << endl; return in;}
    String teststring (const String& in)
      {cout << "String " << in << endl; String out=in; return out;}
    Record testrecord (const Record& in)
      {cout << "Record " << in.nfields() << endl; return in;}
    ValueHolder testvh (const ValueHolder& in)
      {cout << "VH " << in.dataType() << endl; return in;}
    Vector<Int> testvecint (const Vector<int>& in)
      {cout << "VecInt " << in << endl; return in;}
    Vector<DComplex> testveccomplex (const Vector<DComplex>& in)
      {cout << "VecComplex " << in << endl; return in;}
    Vector<String> testvecstr (const Vector<String>& in)
      {cout << "VecStr " << in << endl; return in;}
    IPosition testipos (const IPosition& in)
      {cout << "IPos " << in << endl; return in;}
  };


  void testConvert()
  {
    class_<TConvert> ("tConvert", init<>())
      .def ("testbool",       &TConvert::testbool)
      .def ("testint",        &TConvert::testint)
      .def ("testfloat",      &TConvert::testfloat)
      .def ("testcomplex",    &TConvert::testcomplex)
      .def ("teststring",     &TConvert::teststring)
      .def ("testrecord",     &TConvert::testrecord)
      .def ("testvh",         &TConvert::testvh)
      .def ("testvecint",     &TConvert::testvecint)
      .def ("testveccomplex", &TConvert::testveccomplex)
      .def ("testvecstr",     &TConvert::testvecstr)
      .def ("testipos",       &TConvert::testipos)
      ;
  }

}}


BOOST_PYTHON_MODULE(_tConvert)
{
  casa::appspython::register_convert_excp();
  casa::appspython::register_convert_casa_string();
  casa::appspython::register_convert_casa_vector<casa::String>();
  casa::appspython::register_convert_casa_vector<casa::DComplex>();
  casa::appspython::register_convert_casa_vector<casa::Int>();
  casa::appspython::register_convert_casa_iposition();
  casa::appspython::register_convert_casa_valueholder();
  casa::appspython::register_convert_casa_record();

  casa::appspython::testConvert();
}
